src/views/partials/actor-panel.html.slang

```diff
@@ -17,6 +17,13 @@
       br: a href=actor.display_link = actor.account_uri
       - if (summary = actor.summary.presence)
         br: == s summary
+	.ui.segment
+	  - if (attachments = actor.attachments.presence)
+        table
+          - attachments.each do |attachment|
+            tr
+              th.name == attachment.name
+              td.value == maybe_show_link(attachment.value)
```

Comment:
Wanted to add a helper here to display links as HTML in the HTML and JSON views, so we can store them as plain links. But was not able to figure out where to put it that it would be in scope.

Reply:
converting url strings to html links would be a useful utility function. it could go into something like src/framework/util.cr

Reply:
the other angle is maybe punt on this entirely for the time being. if the values can be html then anyone can author links in html. otherwise, it seems like you have to either convert urls to links when the values are first set, or you have to do it on the fly when they are presented. if the former, then users will have to deal with a URL they entered presented back to them as html when they edit it. if the later, then the link creation will have to be smart enough to, at least, not make urls inside of the href attribute of an existing link into a link and breaking the html.

imo, the right long term solution is probably to implement rich text/html editing for these fields, as well as summary and site fields like the footer. then users can just wysiwyg their profile metadata, summary, etc. etc.

Reply:
The implementation that is in this PR currently has them written and stored as plaintext to make editing easy, and converted to HTML links on presentation. Since Mastodon converts to HTML in the JSON as well, I also did that. If you were to enter HTML in the original input field, it will just return it back without any processing, because the link detector is intentionally only looking for strings that _start_ with what appears to be a proper URL. If you were to put your own HTML link in there, you would get that back unmodified next time you edit. With regard to federation, I feel like that's most easily solved with sanitizing on intake.

Reply:
👍 

one downside to sanitizing on intake is that you lose local access to the original content. (the upside is that you do the sanitizing once.) i'm a data nerd so that's always less attractive.

Reply:
Sounds good. Either way works, if you'd prefer to keep the original content, I will do that. I don't think this is a high volume activity since it would only happen when you view federated profile on Ktistec, so doing it on render seems OK. Just means that we'll always present sanitized HTML, even if it was created locally.

Reply:
you can have it both ways. for objects we have content and media_type, which are activitypub fields, and a source, which is the original (with a media type, as well, in case we support markdown, etc. in the future.

(i'm not arguing for this, btw—i don't know if i'd go to the trouble, myself, but i'm pointing it out in case you find it valuable)